lwtime/nattime.go

```diff
@@ -0,0 +1,198 @@
+	nt := natural{}
+	s = strings.ToLower(s)
+
+	// Today
+	if naturalAdjective(s) == Today {
+		nt.adjective = This
+		nt.num = "1"
+		nt.iNum = 1
+		nt.unit = relativeUnit("d")
+		return nt, nil
+	}
+	// Yesterday
+	if naturalAdjective(s) == Yesterday {
+		nt.adjective = Previous
+		nt.num = "1"
+		nt.iNum = 1
+		nt.unit = relativeUnit("d")
+		return nt, nil
+	}
+	// Singular
+	var nt_parts []string
+	singularRE := regexp.MustCompile(naturalSingularRE)
+	if nt_parts = singularRE.FindStringSubmatch(s); s == "" || nt_parts == nil {
+		// Plural
+		pluralRE := regexp.MustCompile(naturalPluralRE)
+		if nt_parts = pluralRE.FindStringSubmatch(s); s == "" || nt_parts == nil {
+			return nt, errors.New(fmt.Sprintf("natural time (%s) is invalid", s))
+		}
+	}
+	// Adjective
+	nt.adjective = naturalAdjective(nt_parts[1])
+	if !nt.adjective.isValid() {
+		// this would indicate a code mismatch between enumerated adjectives and regex
+		return nt, errors.New(fmt.Sprintf("invalid adjective for natural time (%s)", s))
+	}
+	// Num
+	nt.num = nt_parts[2]
+	var err error
+	nt.iNum, err = strconv.Atoi(nt.num)
+	if err != nil {
+		nt.num = "1"
+		nt.iNum = 1
+	}
+	// Unit
+	if ok := nt.loadRelativeUnit(nt_parts[3]); !ok {
+		// this would indicate a code mismatch between relative units and regex
+		return nt, errors.New(fmt.Sprintf("invalid unit for natural time (%s)", s))
+	}
+	return nt, nil
+}
+
+func (nt natural) range_(t time.Time) (start time.Time, end time.Time, err error) {
+	var relStart, relEnd string
+	baseErr := "unable to compute natural time range"
+
+	// relatives
+	if relStart, relEnd, err = nt.getRelativeRange(); err != nil {
+		err = errors.Wrap(err, baseErr)
+		return
+	}
+
+	// start time
+	if start, err = parseRelativeFromTime(relStart, t); err != nil {
+		err = errors.Wrap(err, baseErr)
+		return
+	}
+
+	// end time
+	if end, err = parseRelativeFromTime(relEnd, t); err != nil {
+		err = errors.Wrap(err, baseErr)
+		return
+	}
+
+	return
+}
+
+func (nt natural) getRelativeRange() (relStart string, relEnd string, err error) {
+	// use natural adjective to determine relative start/end specifiers
+	switch nt.adjective {
+	case This, Current:
+		relStart = fmt.Sprintf("@%s", nt.unit)
+		relEnd = "now"
+	case Previous:
+		relStart = fmt.Sprintf("-1%s@%s", nt.unit, nt.unit)
+		relEnd = fmt.Sprintf("@%s", nt.unit)
+	case Last:
+		relStart = fmt.Sprintf("-%s%s", nt.num, nt.unit)
+		relEnd = "now"
+	default:
+		err = errors.New("invalid adjective for natural time")
+	}
+	return
+}
+
+// Parse the string representation of a Lacework natural time
+//
+// start, end, err := lwtime.ParseNatural("this year")
+// if err != nil {
+// 	...
+// }
+func ParseNatural(n string) (time.Time, time.Time, error) {
+	return parseNaturalFromTime(n, time.Now())
+}
+
+func parseNaturalFromTime(n string, fromTime time.Time) (time.Time, time.Time, error) {
+	var start, end time.Time
+
+	natural, err := newNatural(n)
+	if err != nil {
+		return start, end, err
+	}
```

Comment:
nit: no need to instantiate those start and end variables
```suggestion
	natural, err := newNatural(n)
	if err != nil {
		return time.Time{}, time.Time{}, err
	}
```

Reply:
Fixed.